src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -480,18 +389,15 @@ class LspConnectionManager {
 
     if (!this._compositionHandlersSet) {
       this._initCompositionHandlers()
-      setTimeout(() => {
-        // XXX: wiedld (25 Aug 2022) - cannot init composition until after didOpen file
-        // hardcode a delay for now
-        // TODO(wiedld): https://github.com/influxdata/ui/issues/5305
-        this._initDelayBeforeConsume = false
-        this._incrementBuffer()
-      }, 3000)
+      return // do not re-init schema session on reload
     }
 
     const {toAdd, toRemove} = this._diffSchemaChange(schema, previousState)
     if (Object.keys(toAdd).length || Object.keys(toRemove).length) {
-      this._updateLsp(toAdd, toRemove)
+      // since this._diffSchemaChange() can set the model
+      // we need the executeCommand to be issued after the model update
+      const delay = toAdd.bucket ? 1500 : 0
+      setTimeout(() => this._updateLsp(toAdd, toRemove), delay)
```

Comment:
Can you explain why we need these timeout delays? I was not able to figure it out by following the chain of issues. Maybe a call would be good for this?

Reply:
1. When we start a brand new script, there is [default text populated](https://github.com/influxdata/ui/blob/74b2910b4c94fd4886446732893dc4a11ba79552/src/dataExplorer/components/resources/types/script/index.ts#L26) in the editor.
2. When inserting a new composition for the first time:
   * if there is only the default text present, t[hen remove the default text](https://github.com/influxdata/ui/blob/9c4fb1fc5900cc9c898868b6cd15b75a3f4ba58f/src/languageSupport/languages/flux/lsp/connection.ts#L323-L326).
3. Since the resetting of the model text is indirect (a.k.a. `this._model.setValue('')`), we do introduce a race condition.
    * `this._model.setValue('')` --> monaco-editor updates --> monaco-editor syncs with LSP by sending over the new File.
    * `this._updateLsp()` --> we send to the LSP an `executeCommand` --> LSP makes update using it's current copy of File
4. when dev'ing locally, I occasionally saw this race condition come into play. 

I could:
(a) make this delay occur only when we are removing the default text, if you would like. 
(b) Or never remove the default text; in which case it will remain at the end of the newly constructed composition (and the user will need to delete it manually).

Reply:
I feel like the underlying issue is that we communicate with the LSP in 2 ways:
1. the monaco-editor client messages the LSP
2. the UI itself directly injects messages into the communication stream, bypassing the monaco-editor client.

The reason why we are using the approach # 2 above, is because we could not get the monaco-editor to issue the messages we needed. So we introduced the race condition with the bypass.

Reply:
Thanks for this explanation. This is pretty a pretty frustrating set of circumstances 🤷‍♂️